nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/batch/JSONBatchBlobDecoder.java

```diff
@@ -54,7 +54,9 @@ public Blob getBlobFromJSON(ObjectNode jsonObject) {
             BatchManager bm = Framework.getLocalService(BatchManager.class);
             blob = bm.getBlob(batchId, fileId);
 
-            if (RequestContext.getActiveContext() != null) {
+            final boolean drop = !Boolean.parseBoolean(
+                RequestContext.getActiveContext().getRequest().getHeader(BatchManagerConstants.NO_DROP_FLAG));
```

Comment:
Are we not introducing a new risk of NPE here ?
In the previous code we were checking for NPE first, now we use RequestContext.getActiveContext() before we do the test for Null ...